packer/builder/azure/arm/step_query_vm_test.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Open Technologies, Inc.
+// All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.
+// See License.txt in the project root for license information.
+
+package arm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+
+	"github.com/Azure/packer-azure/packer/builder/azure/common/constants"
+
+	"github.com/mitchellh/multistep"
+)
+
+func TestStepQueryVMShouldFailIfValidateFails(t *testing.T) {
```

Comment:
Good catch.  Change it to query for now, and I'll change it again when I fix the name issue.